lib/WorseReflection/Tests/Inference/if-statement/union_or_else.test

```diff
@@ -0,0 +1,12 @@
+<?php
+
+function foo(Foobar | Barfoo | Bazboo $foobar) {
+
+    if ($foobar instanceof Foobar || $foobar instanceof Barfoo) {
+        wrAssertType('Foobar|Barfoo|Bazboo', $foobar);
```

Comment:
not sure about this. we assert `Foobar` or `Barfoo` - but does that mean that it's _not_ Bazboo? do we care?